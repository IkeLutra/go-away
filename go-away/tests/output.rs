#![allow(dead_code)]

use insta::assert_snapshot;

use go_away::{TypeMetadata, TypeRegistry};

#[derive(TypeMetadata)]
struct MyData {
    field_one: String,
    nested: Nested,
}

#[derive(TypeMetadata)]
struct Nested {
    #[serde(rename = "some_other_name")]
    a_string: String,
    an_int: i64,
}

#[test]
fn test_struct_output() {
    let mut registry = TypeRegistry::new();
    MyData::metadata(&mut registry);

    assert_snapshot!(go_away::registry_to_output(registry));
}
