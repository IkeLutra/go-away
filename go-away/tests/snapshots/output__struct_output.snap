---
source: go-away/tests/output.rs
expression: "go_away::registry_to_output(registry)"

---
type Nested struct {
	AString string `json:"some_other_name"`
	AnInt int `json:"an_int"`
	FulfilmentType FulfilmentType `json:"fulfilment_type"`
}
type MyData struct {
	FieldOne string `json:"field_one"`
	Nested Nested `json:"nested"`
}
type FulfilmentType string

const (
	Delivery FulfilmentType = "Delivery"
	Collection FulfilmentType = "Collection"
)

